docker/Dockerfile

```
@@ -0,0 +1,61 @@
+# Base image from Microsoft Azure with Python 3.12
+FROM mcr.microsoft.com/azureml/minimal-py312-inference:20250310.v1
+
+LABEL maintainer="ErdemOzgen"
+LABEL description="Docker container for PyRIT with JupyterNotebook integration"
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Switch to root user to install packages
+USER root
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    git \
+    curl \
+    wget \
+    build-essential \
+    ca-certificates \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Set up working directory
+WORKDIR /app
+
+# Install PyTorch with CUDA support
+RUN pip install --no-cache-dir torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
+
+# Install JupyterLab and related packages
+RUN pip install --no-cache-dir \
+    jupyterlab \
+    notebook \
+    ipywidgets \
+    matplotlib \
+    pandas \
+    seaborn
+
+# Install PyRIT from PyPI (the official way)
+RUN pip install --no-cache-dir pyrit[all]
```

```suggestion
RUN pip install --no-cache-dir pyrit[dev,all]
```
might sound counterintuitive but the `dev` dependencies include a few that are useful when running specific notebooks (like semantic-kernel)